[CASSANDRA-19991][4.0] Fix flaky TopPartitionsTest#testServiceTopPartitionsSingleTable

netudima opened the pull request:
Ensure that Sampler.samplerExecutor completed sample tasks during a test
Patch by Dmitry Konstantinov; reviewed by TBD for CASSANDRA-19991

netudima committed:
Fix flaky TopPartitionsTest#testServiceTopPartitionsSingleTable

Ensure that Sampler.samplerExecutor completed sample tasks during a test
Patch by Dmitry Konstantinov; reviewed by TBD for CASSANDRA-19991

smiklosovic commented on test/unit/org/apache/cassandra/Util.java:
```diff
@@ -626,12 +627,17 @@ public static void spinAssertEquals(Object expected, Supplier<Object> actualSupp
+        spinAssert(message, equalTo(expected), actualSupplier, timeout, timeUnit);
+    }
+
+    public static <T> void spinAssert(String message, Matcher<T> mather, Supplier<? extends T> actualSupplier, long timeout, TimeUnit timeUnit)
```
@netudima "matcher" instead of "mather"

netudima replied:
thanks, fixed

Commit:
Fix flaky TopPartitionsTest#testServiceTopPartitionsSingleTable

Fix review comment
Patch by Dmitry Konstantinov; reviewed by Stefan Miklosovic,Berenguer Blasi for CASSANDRA-19991
